File: javascript/binary/binary.js
function Binary(binary){
	this.binary = binary;
	if(binary.match(/[a-z]/i) || binary.match(/[2-9]/i)){
		this.binary = 0;
	}
	this.toDecimal = function(){
		return parseInt((this.binary +'').replace(/[^01]/gi, ''),2);
	}
};
module.exports = Binary;

File: ruby/binary/binary.rb
class Binary
	def self.to_decimal(number)
		if number.delete('01') != ''
			raise new ArgumentError
		end
		total = 0;
		(0..number.length-1).each do |n|
			if number[n] == '1'
				total += 2**(number.length-1 -n)
			end
		end
		total.to_s
		total
	end
end
module BookKeeping
	VERSION = 3
end